Extensions/xkit_patches.js

```diff
@@ -967,15 +967,37 @@ XKit.tools.get_blogs = function() {
 				 * @param {String} new_content
 				 */
 				set_content_html: function(new_content) {
-					var content_editor = $('.post-form--form').find('.editor.editor-richtext');
-					if (content_editor.length === 0) {
-						XKit.console.add('ERROR: unable to set content html');
+					if ($(".html-field").css("display") === "none") {
+						var content_editor = $('.post-form--form').find('.editor.editor-richtext');
+						if (content_editor.length === 0) {
+							XKit.console.add('ERROR: unable to set content html');
+							return;
+						}
+						content_editor.focus();
+						content_editor.html(new_content);
+						content_editor.addClass("editor-richtext-has-text");
+						content_editor.blur();
 						return;
 					}
-					content_editor.focus();
-					content_editor.html(new_content);
-					content_editor.addClass("editor-richtext-has-text");
-					content_editor.blur();
+
+					var html_or_markdown = $(".tab-label[data-js-srclabel]").text();
+					if (html_or_markdown === "HTML") {
+						XKit.tools.add_function(function(new_content){
+							var editor_div = document.getElementsByClassName("ace_editor");
+							if (editor_div.length === 1) {
+								var editor = window.ace.edit(editor_div[0]);
+								editor.setValue(new_content);
+								setTimeout(function(){
+									jQuery(".ace_marker-layer").empty();
+								}, 500);
+							}
+						}, true, "", new_content);
```

Comment:
The changes to xkit.js are unnecessary if you use the `addt` argument of `add_function`, replacing the "" with JSON.stringify(new_content) and beginning the function with `var new_content = add_tag`.